.circleci/config.yml

```diff
@@ -0,0 +1,215 @@
+# iOS CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/ios-migrating-from-1-2/ for more details
+#
+version: 2
+jobs:
+
+  build:
+    working_directory: ~/code
+    docker:
+      - image: circleci/android:api-27-alpha
+    environment:
+      JVM_OPTS: -Xmx3200m
+    steps:
+      - checkout
+      - run:
+          name: build the whole project
+          command: bash gradlew build -x test
+
+  unittest:
+    working_directory: ~/code
+    docker:
+      - image: circleci/android:api-27-alpha
+    resource_class: large
+    environment:
+      JVM_OPTS: -Xmx3200m
+
+    steps:
+      - checkout
+
+      - run:
+          name: run unit tests
+          command: |
+            python3 CircleciScripts/run_unittest.py test_results
+      - run:
+          name : check unit test result 
+          command : bash CircleciScripts/check_test_result.sh test_results
+          
+      - store_artifacts:
+          path: test_results
+
+
+  release_tag:
+    working_directory: ~/code
+    docker:
+      - image:  circleci/golang:1.11
+    steps:
+      - checkout
+      - run:
+          name: install github-release
+          command: go get github.com/aktau/github-release          
+      - run:
+          name: release the tag
+          command: |
+            set -x
+            export release_tag=${CIRCLE_TAG}
+            echo "release_tag:$release_tag"
+            export version=$(echo "$release_tag" | sed 's|.*v\([0-9\.]*\).*|\1|')
+            export tagdescription=$(sed -n "/## \[Release $version\]/,/## \[Release [0-9]*\.[0-9]*\.[0-9]\]/p"  CHANGELOG.md | sed '1d' | sed '$d')
+            echo "tagdescription:$tagdescription"
+            export tagname="AWS SDK for Android v$version"
+            echo "tagname:$tagname"
+            echo "token:${GH_TOKEN}"
+            echo "username:${CIRCLE_PROJECT_USERNAME}"
+            echo "repo:${CIRCLE_PROJECT_REPONAME}"
+            echo "$tagdescription" | github-release release  -s ${GH_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME}  -t $release_tag   --name  "$tagname" -d - 
+
+  release_javadoc:
+    docker:
+      - image: circleci/android:api-27-alpha
+    environment:
+      JVM_OPTS: -Xmx3200m     
+    steps:
+      - checkout   
+
+      - run:
+          name: set version
+          command: |
+            set -x
+            export circleci_tag=${CIRCLE_TAG}
+            export version=$(echo "$circleci_tag" | sed 's|.*v\([0-9\.]*\).*|\1|')
+            echo "version:$version"
+            sed  -i   "s/@SDK_VERSION@/$version/"  gradle.properties  
+            cat gradle.properties 
+      - run:
+          name: generate documents
+          command: |
+            bash gradlew generateJavadocs
+            git checkout -- gradle.properties 
+
+      - run:
+          name: copy documents
+          command: |
+            git config --local user.name "${CIRCLE_PROJECT_USERNAME}"
```

Comment:
what is the value for `CIRCLE_PROJECT_USERNAME`?

Reply:
It's a CircleCI build-in variable:  The GitHub or Bitbucket username of the current project.
 In ios project, we are using AWS. Should we use the same configuration here with iOS project?